reminders/plugin_bot.go

```diff
@@ -220,70 +218,33 @@ var cmds = []*commands.YAGCommand{
 				}
 			}
 
-			// Do the actual deletion
-			err = common.GORM.Delete(reminder).Error
-			if err != nil {
-				return nil, err
-			}
-
-			// Check if we should remove the scheduled event
-			currentReminders, err := GetUserReminders(reminder.UserIDInt())
+			// just deleting from database is enough; we need not delete the
+			// scheduled event since the handler will check database
+			_, err = reminder.DeleteG(parsed.Context(), false /* hardDelete */)
 			if err != nil {
 				return nil, err
 			}
 
-			delMsg := fmt.Sprintf("Deleted reminder **#%d**: '%s'", reminder.ID, limitString(reminder.Message))
-
-			// If there is another reminder with the same timestamp, do not remove the scheduled event
-			for _, v := range currentReminders {
-				if v.When == reminder.When {
-					return delMsg, nil
-				}
-			}
-
-			return delMsg, nil
+			return fmt.Sprintf("Deleted reminder **#%d**: '%s'", reminder.ID, CutReminderShort(reminder.Message)), nil
 		},
 	},
 }
 
-func stringReminders(reminders []*Reminder, displayUsernames bool) string {
-	out := ""
-	for _, v := range reminders {
-		parsedCID, _ := strconv.ParseInt(v.ChannelID, 10, 64)
-
-		t := time.Unix(v.When, 0)
-		tUnix := t.Unix()
-		timeFromNow := common.HumanizeTime(common.DurationPrecisionMinutes, t)
-		if !displayUsernames {
-			channel := "<#" + discordgo.StrID(parsedCID) + ">"
-			out += fmt.Sprintf("**%d**: %s: '%s' - %s from now (<t:%d:f>)\n", v.ID, channel, limitString(v.Message), timeFromNow, tUnix)
-		} else {
-			member, _ := bot.GetMember(v.GuildID, v.UserIDInt())
-			username := "Unknown user"
-			if member != nil {
-				username = member.User.Username
-			}
-			out += fmt.Sprintf("**%d**: %s: '%s' - %s from now (<t:%d:f>)\n", v.ID, username, limitString(v.Message), timeFromNow, tUnix)
-		}
-	}
-	return out
-}
-
 func checkUserScheduledEvent(evt *seventsmodels.ScheduledEvent, data interface{}) (retry bool, err error) {
-	// !important! the evt.GuildID can be 1 in cases where it was migrated from the legacy scheduled event system
+	// IMPORTANT: evt.GuildID can be 1 in cases where it was migrated from the
+	// legacy scheduled event system.
 
-	userID := *data.(*int64)
-
-	reminders, err := GetUserReminders(userID)
+	userID := discordgo.StrID(*data.(*int64))
+	reminders, err := models.Reminders(models.ReminderWhere.UserID.EQ(userID)).AllG(context.Background())
 	if err != nil {
 		return true, err
 	}
 
-	now := time.Now()
-	nowUnix := now.Unix()
-	for _, v := range reminders {
-		if v.When <= nowUnix {
-			err := v.Trigger()
+	// TODO: can we move this filtering step into the database query?
```

Comment:
I think we can, but it is probably preferable to minimize non-database refactors (that are not obviously correct) to prevent introducing bugs. Perhaps we can try this refactor in a future version, once the gorm -> sqlboiler transition is complete.